src/Tickets/Commerce/Gateways/Square/Notices/Webhook_Notice.php

```diff
@@ -0,0 +1,207 @@
+<?php
+/**
+ * Square Webhook Notices
+ *
+ * @since TBD
+ *
+ * @package TEC\Tickets\Commerce\Gateways\Square\Notices
+ */
+
+namespace TEC\Tickets\Commerce\Gateways\Square\Notices;
+
+use TEC\Tickets\Commerce\Gateways\Square\Gateway;
+use TEC\Tickets\Commerce\Gateways\Square\Webhooks;
+
+/**
+ * Square Webhook Notice Class
+ *
+ * @since TBD
+ */
+class Webhook_Notice {
+	/**
+	 * Notice slug
+	 *
+	 * @since TBD
+	 *
+	 * @var string
+	 */
+	protected $notice_slug = 'tec-tickets-commerce-square-webhook-notice';
+
+	/**
+	 * Setup hooks for the service provider.
+	 *
+	 * @since TBD
+	 */
+	public function register() {
+		// First check the webhook status if needed
+		$this->maybe_check_webhook_status();
+
+		tribe_notice(
+			$this->notice_slug,
+			[ $this, 'render_notice' ],
+			[
+				'type'     => 'error',
+				'dismiss'  => true,
+				'priority' => 10,
+			],
+			[ $this, 'should_display_notice' ]
+		);
+	}
+
+	/**
+	 * Maybe check webhook status if we haven't checked recently.
+	 *
+	 * @since TBD
+	 */
+	protected function maybe_check_webhook_status() {
+		// If Square gateway is not enabled, don't perform check
+		if ( ! tribe( Gateway::class )->is_enabled() ) {
+			return;
+		}
+
+		// Get the last check time
+		$status = tribe_get_option( Webhooks::$option_webhook_last_check, [] );
+		$last_checked = isset( $status['last_checked'] ) ? (int) $status['last_checked'] : 0;
+
+		// Check if we need to run a check (not checked in the last 12 hours)
+		if ( time() - $last_checked > 12 * HOUR_IN_SECONDS ) {
+			// Run the webhook health check
+			tribe( Webhooks::class )->check_webhook_health();
+
+			// Also check the webhook configuration
+			tribe( Webhooks::class )->check_webhook_configuration();
+		}
+	}
+
+	/**
+	 * Determines if the webhook notice should be displayed
+	 *
+	 * @since TBD
+	 *
+	 * @return bool
+	 */
+	public function should_display_notice() {
+		// If Square gateway is not enabled, don't show the notice
+		if ( ! tribe( Gateway::class )->is_enabled() ) {
```

Comment:
The service locator is used accross methods to locate the Gateway singleton. Just pass it in the constructor instead please